### PR TITLE
Transformations: Move transformation addition into drawer

### LINKED ISSUE #74197: Transformations: Move transformation addition to a drawer
The current tab UI transformations leaves the transformations hanging out in a sea of text.

Screenshot 2023-08-31 at 10 05 35 AM

In this situation for those not familiar with Grafana design language it may be difficult to notice that this is a clickable UI element as it appears to be a text label because of its relative positioning. 

We're using this to track some experiments to see if we can make this look a bit more clickable.

### PR DESCRIPTION
**Pull request by codeincarnate**
**What is this feature?**

This PR moves the addition of new transforms into a drawer component. Additionally, rather than immediately showing the transformation picker, it shows an empty message for adding transformations.

---

The empty state is as follows and is meant to mimic the empty state of a dashboard:

transform-empty

While this does result in an extra click when no transformations have been added it helps improve clarity for users of transformations, especially those who are not familiar with the feature.

---

When selected the following interface is shown:

transform-drawer

---

In addition to working with the transformation redesign it also works with the old transformation interface as well. This looks like the following:

transform-drawer-old

Beyond the addition of the drawer a few UI tweaks have been made:

- The heading is removed from the interface as it's redundant with the tab title and brings the display closer to the _Query_ tab.
- The delete all transformations button is moved to be alongside the add transformation button
- The toggle for _Show Images_ in the transformation redesign has been removed, as there's now more space for these images.

**Why do we need this feature?**

The current transformation interface doesn't have much room in which to show transformations although there are quite a large number. This improves the space to select transformations while also allowing the current visualization and configured transformations to still be seen.

Beyond that, it improves the experience for newer users of by providing guidance on how transformations function.

**Who is this feature for?**

This is for users of transformations, especially users less familiar with transformations.

**Which issue(s) does this PR fix?**:

Fixes #74197

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by oscarkilhed — COMMENTED** [review]
I think this looks good, especially the added space for transformations.

I'm thinking maybe this should be part of the transformations redesign feature toggle?

File: public/app/features/dashboard/components/TransformationsEditor/TransformationsEditor.test.tsx [hunk 1]
```diff
@@ -24,11 +24,15 @@ describe('TransformationsEditor', () => {
     function renderList() {
       setup();
 
-      const cards = screen.getAllByTestId(/New transform/i);
-      expect(cards.length).toEqual(standardTransformersRegistry.list().length);
+      const addButton = screen.getAllByTestId(selectors.components.Transforms.addTransformationButton + 'i');
+      const emptyMessage = screen.getAllByTestId(selectors.components.Transforms.noTransformationsMessage);
+
+      console.log({ addButton, emptyMessage });
```
Comment (mdvictor) [inline public/app/features/dashboard/components/TransformationsEditor/TransformationsEditor.test.tsx:30]: think this can be removed, right?

**Comment by codeincarnate** [comment]
Getting back to this one. Although we've been doing a lot behind the transformation redesign flag I think we should try to remove it as soon as possible. Unless we have signal that the redesign isn't functioning as intended it's becoming increasingly difficult to correctly separate out the old and new interfaces. Not a deal-breaker by any means. However, this is likely taking longer than it would otherwise.

**Comment by codeincarnate** [comment]
Also, @catherineymgui would be very interested hear your thoughts on this one 😄

**Comment by catherineymgui** [comment]
I think this helps solve the space issue! But can see the drawer interaction being confusing to navigate for current users (slightly unexpected behaviour but I think that's okay as people get used to it).

However, would there be a way to resize the drawer, since it currently is still somewhat hiding the panels / data underneath?

If you haven't already, I would share this with some internal users as I also wonder if the overlay shows has too little contrast (ex: Leon, Sam J, Chris Shih)

**Comment by codeincarnate** [comment]
@catherineymgui Currently drawer isn't resizable but I wouldn't be against adding it. I'll also show around and see what folks think 😃

**Comment by codeincarnate** [comment]
Looked at this further, full re-sizing might be tricky. We're using [rc-drawer](https://github.com/react-component/drawer) and it's looked not maintained. That being said seems like it might be overkill anyone.

Was thinking of adding a control to the drawer that can switch between sizes, there are currently 3 defined at 25%, 50%, and 75% viewport heights so that seems like that would cover it in this case. Definitely thoughts appreciated on that idea too

**Comment by codeincarnate** [comment]
https://github.com/grafana/grafana/assets/199847/8ccc3904-871f-4d80-a839-d73e1749bcd8

OK, this is what I have now. Thoughts?

**Comment by catherineymgui** [comment]
Ooh that could work. I would use the angle icons or the movepane icons instead of the arrows though. Also the recording shows two sizes. Are you planning on having 2 or 3? If 2 sizes - the icon could change from the collapse to expanded depending on the main action.

Explore has something similar although I think the icon has a bug 
Screenshot 2023-11-19 at 20 19 32

**Comment by codeincarnate** [comment]
Nice, there are three levels just wanted to show the two for demonstration. I'll get another video up in that regard

**Comment by codeincarnate** [comment]
@torkelo noticed that we had an expandable feature in the drawer previously to have the drawer show more content. Above is potentially for the opposite, to show more context (or show more in the drawer as necessary). That being said since it was removed would be curious to get your thoughts on having expand/collapse controls in the drawer. CC @catherineymgui

**Comment by codeincarnate** [comment]
I think for now I'm going to separate out the expand/collapse code into a separate PR so as not to block this regardless

**Comment by codeincarnate** [comment]
Closing this one in favor of https://github.com/grafana/grafana/pull/78299

**Pull request closed without merging**